.github/workflows/CI.yml

```diff
@@ -0,0 +1,97 @@
+name: CI
+
+on: [push, pull_request, workflow_dispatch]
+
+jobs:
+  spindle-serial-ubuntu:
+    name: Testsuite (Serial, Ubuntu)
+    environment: Spindle CI
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    steps:
+      - name: Check out Spindle
+        uses: actions/checkout@v5
```

Comment:
To minimize the chance of external modifications creating a failure state in CI, I typically recommend that GitHub projects lock their actions to a specific commit and then use GitHub Dependabot to automatically bump that commit periodically.

https://github.com/LLNL/Spindle/blob/d5e582f316e03f23afcf6f410b06566ebade5f3c/.github/workflows/style.yml#L12

https://github.com/LLNL/Spindle/blob/d5e582f316e03f23afcf6f410b06566ebade5f3c/.github/dependabot.yml#L1-L6